payjoin-relay/src/main.rs

```diff
@@ -259,7 +267,7 @@ async fn post_fallback_v1(
     id: &str,
     query: String,
     body: Body,
-    pool: DbPool,
+    pool: Arc<Mutex<DatabasePool>>,
```

Comment:
DbPool encapsulated `Arc<Mutex<...` details, this is a sign of a leaky abstraction